sdk/src/auth.rs

```diff
@@ -0,0 +1,92 @@
+use std::borrow::Cow;
+use std::str::FromStr;
+
+use alloy::signers::{wallet::LocalWallet, Signer};
+use alloy::sol;
+use alloy::sol_types::{Eip712Domain, SolStruct};
+use anyhow::Result;
+
+sol! {
+    struct CreateProof {
+        uint64 deadline;
+    }
+
+    struct SubmitProof {
+        string proof_id;
+    }
+
+    struct RelayProof {
+        string proof_id;
+        uint32 chain_id;
+        address verifier;
+        address callback;
+        bytes callback_data;
+    }
+}
+
+pub struct NetworkAuth {
+    wallet: LocalWallet,
+}
+
+impl NetworkAuth {
+    pub fn new(private_key: &str) -> Self {
+        let wallet = LocalWallet::from_str(private_key).unwrap();
+        Self { wallet }
+    }
+
+    fn get_domain_separator() -> Eip712Domain {
+        Eip712Domain {
+            name: Some(Cow::Borrowed("succinct")),
+            version: Some(Cow::Borrowed("1")),
+            ..Default::default()
+        }
+    }
+
+    pub async fn sign_create_proof_message(&self, deadline: u64) -> Result<Vec<u8>> {
+        let domain_seperator = Self::get_domain_separator();
+
+        let type_struct = CreateProof { deadline };
+
+        let message_hash = type_struct.eip712_signing_hash(&domain_seperator);
+        let signature = self.wallet.sign_hash(&message_hash).await?;
+
+        Ok(signature.as_bytes().to_vec())
+    }
+
+    pub async fn sign_submit_proof_message(&self, proof_id: &str) -> Result<Vec<u8>> {
+        let domain_seperator = Self::get_domain_separator();
+
+        let type_struct = SubmitProof {
+            proof_id: proof_id.to_string(),
+        };
```

Comment:
nit: Add docs for `NetworkAuth`, as this will be commonly used & it's important to understand what's being signed.

Reply:
added